sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePartitionedUploader.cs

```diff
@@ -278,6 +268,9 @@ internal class DataLakePartitionedUploader
                         cancellationToken: cancellationToken);
 
                     appendedBytes += block.Length;
+
+                    // Dispose the block after the loop iterates and return its memory to our ArrayPool
+                    block.Dispose();
```

Comment:
try/finally might be more analogous to the previous version.

Reply:
Fixed